oncoact/patientreporter/reporting_pipeline/share_report.py

```diff
@@ -51,12 +44,15 @@ def share_report(self, publish_to_portal, notify_users):
         self._delete_old_artifacts_in_portal_bucket()
         self._delete_old_artifacts_in_gcp_share_bucket()
 
-        self._copy_files_to_remote_buckets(publish_to_portal=publish_to_portal)
+        publish_to_portal_and_notify = self.rest_client.get_lama_patient_reporter_data(self.report_created_record['barcode'])[
+            'reportSettings']['isSharedThroughPortal']
```

Comment:
When doing chained access like this, it is important to verify every nested field exists, for example, this code assumes that the reporter_data from lama always contains a 'reportSettings' object with a nested 'isSharedThroughPortal' field.

Maybe it is smart to double check if this is always the case. If so, then this code is perfect as is, but it can often lead to confusing errors when sometimes the nested fields are undefined.